### DISCUSSION TITLE
Set value and display of a MUI Autocomplete from a watched field

### DISCUSSION THREAD

**Discussion by Kalgros20** [react-hook-form/react-hook-form#5241] [Q&A] [answered]
HI folks, i have a huge form, and a lot of MUI autocomplete fields... So, what's the point? Some of these autocomplete fields must be filled dependending of values from another fields. The problem is:  the autocompletes are not updating their values on the screen; The input value is filled correctly.  I know that setValue don't trigger a render, and this could be the reason that value displayed in field is not updated... So my question is, what is the right approach?

Here's a  [sandbox](https://codesandbox.io/s/react-hook-form-nestedvalue-forked-o9kco?file=/src/App.tsx) using Onchange prop.

Here's a [sandbox](https://codesandbox.io/s/react-hook-form-nestedvalue-forked-3c4k4?file=/src/App.tsx) using watch.

Obs: I searched a little bit about the usage of AutoComplete, and in some issues is recommended use the Controller wrapper, but this approach needs the defaultValue... And this breaks a  my design :(   The sandbox, isn't using the Controller, but if the only solution is use, i can think about it.

**Comment by Kalgros20** [comment, marked as answer]
I was able to do guys using reset API.

Here's my solution.

[Code Sandbox](https://codesandbox.io/s/react-hook-form-v6-controller-forked-pgsds?file=/src/index.js)

I know that it's not perfect, but i was able to do what i wanted... that is change de default value of AutoComplete. :)

**Reply by bluebill1049** [reply]
glad you work it out, thanks for sharing the solution! ❤️
